products/sentinel_one.py

```diff
@@ -660,46 +664,49 @@ def _run_query(self, merged_query: str, start_date: datetime, end_date: datetime
                 events = self._get_dv_events(query_id, p_bar_needed=p_bar_needed, cancel_event=cancel_event)
             self.log.debug(f'Got {len(events)} events')
 
-            self._results[merged_tag] = list()
-            
-            for event in events:
-                if self._pq:
-                    hostname = event[0]
-                    username = event[1]
-                    path = event[2]
-                    command_line = event[3]
-                    additional_data = (event[8], event[9], event[10], event[11],'None','None','None','None','None','None','None','None','None','None','None','None')
-                else:
-                    hostname = event['endpointName']
-                    username = event['srcProcUser']
-                    path = event['srcProcImagePath']
-                    srcprocstorylineid = event['srcProcStorylineId'] if 'srcProcStorylineId' in event else 'None'
-                    srcprocdisplayname = event['srcProcDisplayName'] if 'srcProcDisplayName' in event else 'None'
-                    tgtprocdisplayname = event['tgtProcDisplayName'] if 'tgtProcDisplayName' in event else 'None'
-                    tgtfilepath = event['tgtFilePath'] if 'tgtFilePath' in event else 'None'
-                    tgtfilesha1 = event['fileSha1'] if 'fileSha1' in event else 'None'
-                    tgtfilesha256 = event['fileSha256'] if 'fileSha256' in event else 'None'
-                    scrprocparentimagepath = event['srcProcParentImagePath'] if 'srcProcParentImagePath' in event else 'None'
-                    tgtprocimagepath = event['tgtProcImagePath'] if 'tgtProcImagePath' in event else 'None'
-                    url = event['networkUrl'] if 'networkUrl' in event else 'None'
-                    srcip = event['srcIp'] if 'srcIp' in event else 'None'
-                    dstip = event['dstIp'] if 'dstIp' in event else 'None'
-                    dnsrequest = event['dnsRequest'] if 'dnsRequest' in event else 'None'
-                    command_line = event['srcProcCmdLine']
-                    additional_data = (event['eventTime'], event['siteId'], event['siteName'], srcprocstorylineid, srcprocdisplayname, scrprocparentimagepath, tgtprocdisplayname, tgtprocimagepath, tgtfilepath, tgtfilesha1, tgtfilesha256, url, srcip, dstip, dnsrequest, event['eventType'])
-
-                result = Result(hostname, username, path, command_line, additional_data)
-                
-                # Raw Feature (Inactive)
-                '''
-                if self._raw:
-                    self._results[merged_tag].append(event)
-                else:
-                    self._results[merged_tag].append(result)
-                '''
-
-                self._results[merged_tag].append(result)
+            if self._json:
+                self._results[merged_tag] = dict()
+                self._results[merged_tag] = events
```

Comment:
Can the code be configured to use the existing `Result` class? The type defined for `self._results[merged_tag]` is a list of `Result` objects rather than a dictionary

Reply:
this is going to be tricky. since the Result class uses specific data fields that won't be filled in when just dumping the raw json response. the response is not parsed when using the --json option and is meant to output the raw data object within the json response. With this method an analyst, could import the raw json into a SIEM or other tool to perform additional analysis.